nix/overlay.nix

```diff
@@ -0,0 +1,34 @@
+final: prev:
+let
+  inherit (final) lib;
+  static_stdenv = final.stdenv.hostPlatform.isMusl;
+in {
+  nix-gitDescribe = final.callPackage ./gitDescribe.nix {};
+  arrow-cpp = prev.arrow-cpp.overrideAttrs (old: {
+    patches = old.patches ++ lib.optional static_stdenv arrow/fix-static-jemalloc.patch;
+  });
+  caf = let
+    source = builtins.fromJSON (builtins.readFile ./caf/source.json);
+    in prev.caf.overrideAttrs (old: {
+    # fetchFromGitHub uses ellipsis in the parameter set to be hash method
+    # agnostic. Because of that, callPackageWith does not detect that sha256
+    # is a required argument, and it has to be passed explicitly instead.
+    src = lib.callPackageWith source final.fetchFromGitHub { inherit (source) sha256; };
+  } // lib.optionalAttrs static_stdenv {
+    cmakeFlags = old.cmakeFlags ++ [
+      "-DCAF_BUILD_STATIC=ON"
+      "-DCAF_BUILD_STATIC_ONLY=ON"
+      "-DOPENSSL_USE_STATIC_LIBS=TRUE"
+      "-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=ON"
+      "-DCMAKE_POLICY_DEFAULT_CMP0069=NEW"
+    ];
+    hardeningDisable = [
+      "pic"
+    ];
+    NIX_CFLAGS_COMPILE = "-std=c++17";
```

Comment:
Do we really need to explicitly set this here again? That's another point where we can diverge from CMake.

Reply:
Please note that this is for building CAF + VAST in static mode with `-flto`. The compiler complains if it sees different definitions of the same types in its internal passes. This can be removed once we migrate to CAF 0.18.

Reply:
```suggestion
    # TODO: Remove when updating to CAF 0.18.
    NIX_CFLAGS_COMPILE = "-std=c++17";
```

Reply:
If we're going to leave a comment, let's also mention *why* this can be removed when updating to CAF 0.18, otherwise people in the future will have to reverse-engineer this conversation.